apps/builder/app/shared/sync/sync-stores.ts

```diff
@@ -303,4 +303,10 @@ export const useBuilderStore = (publish: Publish) => {
       };
     });
   }, [publish]);
+
+  useEffect(() => {
+    return () => {
+      handshakenStore.set(false);
```

Comment:
I thinking maybe reset handshake in the beginning instead?

Reply:
will be dependent on order of useEffects

Reply:
Also some atoms initiated before, and here can be all project dependent cleanup if we will not find any context solution